src/OrleansRuntime/Placement/IActivationSelector.cs

```diff
@@ -9,10 +10,13 @@ internal interface IActivationSelector
     {
         Task<PlacementResult> OnSelectActivation(
             PlacementStrategy strategy, GrainId target, IPlacementRuntime context);
+
+        bool TrySelectActivationSynchronously(
```

Comment:
In that case, you can have a single method on the API and implement it with a fast-path branch returning synchronously and an async branch which calls an async method and wraps its Task into a ValueTask. If we raise the C# level to 7.0 then we can use a local function for that or make use of async for Task-like types.